src/Kafka/Producer.purs

```diff
@@ -49,6 +50,23 @@ derive instance ordAcks :: Ord Acks
 
 type AcksImpl = Int
 
+data CompressionType
+  = CompressionTypeNone
+  | CompressionTypeGzip
+
+derive instance eqCompressionType :: Eq CompressionType
+derive instance ordCompressionType :: Ord CompressionType
+
+-- | https://github.com/tulios/kafkajs/blob/dcee6971c4a739ebb02d9279f68155e3945c50f7/types/index.d.ts#L1116
+-- | `CompressionTypes`
+-- |
+-- | https://github.com/tulios/kafkajs/blob/dcee6971c4a739ebb02d9279f68155e3945c50f7/src/protocol/message/compression/index.js#L5
+-- | `Compression.Types`
+-- |
+-- | NOTE only `GZIP` is implemented
+-- | see [`Compression.Codecs`](https://github.com/tulios/kafkajs/blob/dcee6971c4a739ebb02d9279f68155e3945c50f7/src/protocol/message/compression/index.js#L13-L24)
```

Comment:
If only gzip is supported, then why have `CompressionType` at all? Can we just assume gzip? Or if the producer records originate externally, then shouldn't we just throw an error immediately if the value isn't `1`?

Reply:
Turns out GZIP is natively supported by Kafka while other compression codecs are provided by separate packages like [`kafkajs-snappy`](https://kafka.js.org/docs/producing#a-name-compression-snappy-a-snappy). So I'll complete the enum and make a comment on package installation.

Reply:
Oh, I misunderstood the comment, I thought the only two options were none and gzip, and only gzip was supported, which I guess doesn't make sense, as there's no reason no compression wouldn't be supported. Now that I know there are more compression options, that comment made more sense.

If we were only planning to use this library internally I would say no change was needed, but I guess the goal is to open a PR with the original library, or at the least, maintain this as an open source library, so then supporting the other options makes sense.

Reply:
I have this being an open sourced library in mind when designing the public API. Comments are updated to suggest how to use compression codecs other than gzip.